mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIUploadHelper.java

```diff
@@ -154,27 +160,55 @@ public static Response uploadDerivate(UriInfo info, HttpServletRequest request,
             MCRObjectID derID = null;
-            if (overwriteOnExistingLabel) {
-                for (MCRMetaLinkID derLink : mcrObj.getStructure().getDerivates()) {
-                    if (label.equals(derLink.getXLinkLabel()) || label.equals(derLink.getXLinkTitle())) {
-                        derID = derLink.getXLinkHrefID();
+            if (overwriteOnExisting) {
+                if (label != null && label.length() > 0) {
+                    for (MCRMetaLinkID derLink : mcrObj.getStructure().getDerivates()) {
+                        if (label.equals(derLink.getXLinkLabel()) || label.equals(derLink.getXLinkTitle())) {
+                            derID = derLink.getXLinkHrefID();
+                        }
+                    }
+                }
+                if (classification != null && classification.length() > 0) {
+                    MCRCategoryID categid = MCRCategoryID.fromString(classification);
+                    if (MCRCategoryDAOFactory.getInstance().exist(categid)) {
+                        for (MCRMetaEnrichedLinkID derLink : mcrObj.getStructure().getDerivates()) {
+                            for (Content c : derLink.getContentList()) {
+                                if (c instanceof Element && ((Element) c).getName().equals("classification")) {
+                                    Element e = (Element) c;
+                                    if (categid.getRootID().equals(e.getAttributeValue("classid"))
+                                            && categid.getRootID().equals(e.getAttributeValue("categid"))) {
+                                        derID = derLink.getXLinkHrefID();
+                                    }
+                                }
+                            }
+                        }
                     }
                 }
             }
 
             if (derID == null) {
                 derID = MCRObjectID.getNextFreeId(mcrObjIDObj.getProjectId() + "_derivate");
                 MCRDerivate mcrDerivate = new MCRDerivate();
-                mcrDerivate.setLabel(label);
+                if (label != null && label.length() > 0) {
+                    mcrDerivate.setLabel(label);
+                }
                 mcrDerivate.setId(derID);
                 mcrDerivate.setSchema("datamodel-derivate.xsd");
                 mcrDerivate.getDerivate().setLinkMeta(new MCRMetaLinkID("linkmeta", mcrObjIDObj, null, null));
                 mcrDerivate.getDerivate()
-                    .setInternals(new MCRMetaIFS("internal", UPLOAD_DIR.resolve(derID.toString()).toString()));
+                        .setInternals(new MCRMetaIFS("internal", UPLOAD_DIR.resolve(derID.toString()).toString()));
+
+                if (classification != null && classification.length() > 0) {
+                    MCRCategoryID categid = MCRCategoryID.fromString(classification);
+                    if (MCRCategoryDAOFactory.getInstance().exist(categid)) {
```

Comment:
We should throw a error if you pass a not null classification which does not exist.

Reply:
done